Create are_ids_eq.mdx (#276)
comcode-org/hackmud_wiki: docs/scripting/scripts.lib/are_ids_eq.mdx

---
title: scripts.lib().are_ids_eq()
---

### Description

> Compares two strings and evaluates if they are equal. Useful for comparing MongoDB Object IDs.

Reviewer: ```suggestion
Compares two strings or ObjectIDs and evaluates if they are equal. Useful for comparing MongoDB documents by IDs.
```

## Syntax

```js
return #fs.scripts.lib().are_ids_eq("65c982ee9edd1f997c9806ac", "65c982ee9edd1f997c9806a0");
```

### Parameters

#### id1

> The first string value.

Reviewer: ```suggestion
The first string or ObjectID value.
```

#### id2

> The second string value.

Reviewer: ```suggestion
The second string or ObjectID value.
```

### Return

Returns a boolean.

```
False
```

## Example

```js
function(context, args) {
  let l = #fs.scripts.lib();
  let id1 = "65c982ee9edd1f997c9806ac";
  let id2 = "65c982ee9edd1f997c9806a0";
  return l.are_ids_eq(id1, id2);
}

> ```

Reviewer: ````suggestion
```

```js
function(context, args) {
  let l = #fs.scripts.lib();
  let id1 = { $oid: "65c982ee9edd1f997c9806ac" };
  let id2 = { $oid: "65c982ee9edd1f997c9806a0" };
  return l.are_ids_eq(id1, id2);
}
```
````
